### PR TITLE
Wrapper API created

### PR DESCRIPTION
**Pull request by GonzRodriguez**
In first place apologies for taking so long. I had to catch up on jest and typescript and as I said, I have to do this on my spare time. 

I've created de wrapper API and I think this is what you were looking for but please let me know if I'm wrong. 

The second thing I wanted to discuss is testing. I've read that the tests shouldn't hit the API. In that case I really don't know what to do to test it because in my point of view that's the only thing the program does. 

Thank for your patience!

### CODE REVIEW

File: babel.config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "presets": [
+    "@babel/preset-env"
```
Comment (sondreb) [inline babel.config.json:3]: Could you explain why this is included?

**Review by sondreb — CHANGES_REQUESTED** [review]
Looks good so far, but not sure how to run the vscode task properly. Running "npm run test" works fine, but running the launch.json task exits with: "No tests found, exiting with code 1".

**Comment by sondreb** [comment]
> In first place apologies for taking so long. I had to catch up on jest and typescript and as I said, I have to do this on my spare time.
> 
> I've created de wrapper API and I think this is what you were looking for but please let me know if I'm wrong.
> 
> The second thing I wanted to discuss is testing. I've read that the tests shouldn't hit the API. In that case I really don't know what to do to test it because in my point of view that's the only thing the program does.
> 
> Thank for your patience!

Great and thanks for the PR!

There are multiple ways we could test the API. I made the constructor allow override of baseUrl. This way we can host a temporary http server with integration testing. But we should also have the ability to run tests against the deployed APIs.

**Comment by sondreb** [comment]
Here is a good example where you could follow DRY (Don't Repeat Yourself):

image

Most of this code is the same, so you could optimize it a bit by having an "executeGet" method that takes the relative URL, and perhaps some other parameters, depending on what you need.

File: babel.config.json (hunk 1 above)
Comment (GonzRodriguez) [reply babel.config.json:3]: By mistake I think. I've been following tutorials and courses and I believe I installed it by a misunderstanding and forgot to delete it.

**Comment by GonzRodriguez** [comment]
Thanks for the quick response. 
I'm reviewing it now. 
First of all I will make the code more concise. 
After that

**Comment by GonzRodriguez** [comment]
Thanks for the quick response.
I'm reviewing it now.

First of all I will make the code more concise.

After that I will make the tests that actually hits the API as you said.

**Review by sondreb — APPROVED** [review]
Approved with the latest changes! Fantastic work, thanks a lot!

**Comment by sondreb** [comment]
Closes #1

**Pull request merged**